main.rb

```diff
@@ -0,0 +1,166 @@
+require_relative 'person'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def run
+    puts 'Welcome to School Library App!'
+
+    loop do
+      print_options
+
+      option = gets.chomp
+
+      break if option == '7'
+
+      handle_option option
+    end
+
+    puts 'Thank you for using this app!'
+  end
+
+  private
+
+  def handle_option(option)
+    case option
+    when '1'
+      list_books
+    when '2'
+      list_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals_by_person_id
+    else
+      puts 'That is not a valid option ❌'
+    end
+  end
+
+  def print_options
+    puts
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book '
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def list_books
+    @books.each { |book| puts book }
+  end
+
+  def list_people
+    @people.each { |person| puts person }
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2) ❓ [Input the number]: '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'That is not a valid input ❌'
+      return
+    end
+
+    puts 'Person created successfully ✅'
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission ❓ [Y/N]: '
+    parent_permission = gets.chomp.downcase == 'y'
+
+    student = Student.new(name: name, age: age, parent_permission: parent_permission, classroom: @classroom)
```

Comment:
- [ ] This method is throwing an error while I try to create a student. Kindly fix this issue. :pray: 

![Screenshot from 2021-11-13 00-34-07](https://user-images.githubusercontent.com/64894927/141537614-01369c2f-1127-4f79-bd75-6b69d1a90949.png)

Reply:
This is working perfectly.
![student](https://user-images.githubusercontent.com/44624138/141540011-72af29fb-c55b-45bc-8d09-12765b99bbe7.jpg)

Reply:
This issue is well resolved! ✔️